compiler/zrc_parser/src/lexer.rs

```diff
@@ -122,7 +124,11 @@ fn lex_string_contents<'input>(
             Ok(token_inner) => tokens.push(token_inner),
             Err(()) => {
                 return Err(InternalLexicalError::UnknownEscapeSequence(
-                    Span::from_positions(span.start + start_offset, span.end + start_offset),
+                    Span::from_positions_and_file(
```

Comment:
Is it possible to pass the file name onto the lexer for lexical errors to have the correct file span? Not sure if Logos supports this